ipn/ipnlocal/local.go

```diff
@@ -253,14 +253,20 @@ func (b *LocalBackend) UpdateStatus(sb *ipnstate.StatusBuilder) {
 	}
 }
 
-// WhoIs reports the node and user who owns the node with the given IP.
+// WhoIs reports the node and user who owns the node with the given IP:port.
+// If the IP address is unique to this node, the port provided may be 0.
 // If ok == true, n and u are valid.
-func (b *LocalBackend) WhoIs(ip netaddr.IP) (n *tailcfg.Node, u tailcfg.UserProfile, ok bool) {
+func (b *LocalBackend) WhoIs(ipport netaddr.IPPort) (n *tailcfg.Node, u tailcfg.UserProfile, ok bool) {
 	b.mu.Lock()
 	defer b.mu.Unlock()
-	n, ok = b.nodeByAddr[ip]
+	n, ok = b.nodeByAddr[ipport.IP]
 	if !ok {
-		return nil, u, false
+		if ipport.Port != 0 {
+			n, ok = b.e.WhoIsIPPort(ipport)
```

Comment:
yeah, looks like you only care about the userid. So let's make WhoIsIPPort return a tailcfg.UserID instead (a value type, not a pointer) so we don't need to worry about ownership/corruption/data races from potential mutations.

Reply:
I think we also need the node itself? The userid is used to get `u` but this function returns both `u` and `n`, and `n` is a Node. is there a way to get that here from the userid?

Reply:
look 6 lines down

Reply:
oh, for the node, you can use nodeByAddr as the lookup... return the Tailscale IP instead?